test/generator/sancov_aottest.cpp

```diff
@@ -0,0 +1,116 @@
+#include "HalideBuffer.h"
+#include "HalideRuntime.h"
+#include "sancov.h"
+
+#include <iostream>
+#include <limits>
+#include <type_traits>
+#include <vector>
+
+using namespace std;
+using namespace Halide::Runtime;
+
+bool enable_callbacks = false;
+
+#if defined(__linux__)
+// Used by -fsanitize-coverage=stack-depth to track stack depth
+__attribute__((tls_model("initial-exec"))) thread_local uintptr_t __sancov_lowest_stack;
+#endif
+
+extern "C" void __sanitizer_cov_8bit_counters_init(uint8_t *Start, uint8_t *Stop) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_8bit_counters_init. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_pcs_init(const uintptr_t *pcs_beg,
+                                         const uintptr_t *pcs_end) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_pcs_init. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_trace_cmp1(uint8_t Arg1, uint8_t Arg2) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_trace_cmp1. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_trace_cmp4(uint32_t Arg1, uint32_t Arg2) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_trace_cmp4. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_trace_cmp8(uint64_t Arg1, uint64_t Arg2) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_trace_cmp8. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_trace_const_cmp1(uint8_t Arg1, uint8_t Arg2) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_trace_const_cmp1. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_trace_const_cmp2(uint16_t Arg1, uint16_t Arg2) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_trace_const_cmp2. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_trace_const_cmp4(uint32_t Arg1, uint32_t Arg2) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_trace_const_cmp4. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_trace_const_cmp8(uint64_t Arg1, uint64_t Arg2) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_trace_const_cmp8. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_trace_switch(uint64_t Val, uint64_t *Cases) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_trace_switch. Success!\n");
+}
+
+extern "C" void __sanitizer_cov_trace_pc_indir(uintptr_t Callee) {
+    if (!enable_callbacks) return;
+    printf("Hit __sanitizer_cov_trace_pc_indir. Success!\n");
+}
+
+template<typename T>
+void clear_out(T &image) {
+    image.fill(-42);
```

Comment:
Nit: if `T` is an unsigned type, some compilers can complain about assigning a signed literal to it. Either avoid using a negative value or (probably more simply) just always use a signed type here.